functionary/ui/views/function.py

```diff
@@ -73,7 +74,9 @@ def execute(request: HttpRequest) -> HttpResponse:
 
     data: QueryDict = request.POST.copy()
     func = get_object_or_404(Function, id=data.get("function_id"))
-    form = TaskParameterForm(func, data)
+    handle_file_parameters(env, request)
```

Comment:
It seems like this ought to happen after the `form.is_valid()` call for sure.  Otherwise we're uploading files that may never get used because the other tasking input wasn't valid.

I might even say to put it after the `task.clean()` but before the `task.save()`.  That way the file upload is the last thing to happen before the tasking would actually get kicked off.

Reply:
Fixed